### PR TITLE
proposal-execution-git-hub

### PR DESCRIPTION
**Pull request by shivamgarglab49**

### CODE REVIEW

File: utils/EventConsumer.ts [hunk 1]
```diff
@@ -78,31 +59,62 @@ export class EventConsumer {
...
+            eventAbi.anonymous === true ? log.topics.splice(1) : log.topics
           );
-          this.logRecords.push(event);
-        });
-      })
-      .catch(function (err) {
-        console.error(err);
-      });
+          const items = eventsResult.get(eventAbi.name) ?? [];
+          items.push(decodedLog);
```
Comment (amandeeprehal-lab49) [inline utils/EventConsumer.ts:113]: Pls optimize this.

File: utils/EventConsumer.ts [hunk 2]
```diff
@@ -1,44 +1,25 @@
...
-    );
-    const decodedLog = this.web3.eth.abi.decodeLog(
-      eventAbi.inputs,
-      log,
-      topics
-    );
-    return decodedLog;
+  public getSignatureAndEventEntries() {
+    const details: any[] = [];
+    for (const entry of Array.from(this.eventSignatureToNameMap.entries())) {
```
Comment (amandeeprehal-lab49) [inline utils/EventConsumer.ts:17]: We use map directly, pls simplify it.

File: deployment/scripts/proposalScript.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      (await governor.claimGODToken(proposalId, contractId!));
+    console.log("------- exectuion completed:", proposalId);
+  } catch (e) {
+    console.error("- failed to execute propsal:", proposal.proposalId, e);
+  }
+}
+
+async function executeProposals(proposals: any[]) {
+  console.log("- proposals execution flow started");
+  for (let index = 0; index < proposals.length; index++) {
```
Comment (amandeeprehal-lab49) [inline deployment/scripts/proposalScript.ts:53]: Pls change -

 for (const contract of contracts)

File: utils/EventConsumer.ts [hunk 4]
```diff
@@ -78,31 +59,62 @@ export class EventConsumer {
...
+  ) {
+    console.log(`- Getting event(s) from mirror`);
+    if (delayRequired) {
+      const delay = (ms: any) => new Promise((res) => setTimeout(res, ms));
+      console.log(`- Waiting 10s to allow transaction propagation to mirror`);
+      await delay(10000);
+    }
+    const url = `https://testnet.mirrornode.hedera.com/api/v1/contracts/${contractId}/results/logs?order=desc`;
+    console.log("- Request url:", url);
+    const response = await axios.get(url);
```
Comment (amandeeprehal-lab49) [inline utils/EventConsumer.ts:74]: Can we Mock axios?

Comment (shivamgarglab49) [reply utils/EventConsumer.ts:72]: will check separately.

File: deployment/scripts/eventReader.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+const contractMetadata = new ContractMetadata();
+const contractService = new ContractService();
+const helper = new Helper();
+
+export async function main(_contractId: string? = null) {
+  const contractId =
+    _contractId ?? process.env.PROPOSAL_CONTRACT_ID ?? undefined;
+  let contract: DeployedContract;
+  if (contractId !== undefined) {
```
Comment (amandeeprehal-lab49) [inline deployment/scripts/eventReader.ts:15]: I think, you missed this.

File: deployment/scripts/proposalScript.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+export async function main() {
+  contractId = process.env.PROPOSAL_CONTRACT_ID;
+  if (!contractId) {
+    throw Error("Proposal contract id missing.");
+  }
+  const events = await eventReader(contractId);
+  const proposalsExecuted = events.get("ProposalExecuted") ?? [];
+  const proposalsCanceled = events.get("ProposalCanceled") ?? [];
+  const proposalsCreated = events.get("ProposalCreated") ?? [];
+  const proposalsActive = removeSubset(proposalsCreated, [
```
Comment (amandeeprehal-lab49) [inline deployment/scripts/proposalScript.ts:19]: activeProposals make more sense

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**